File: android/ValleTPV/app/src/main/java/com/valleapp/valletpv/adaptadoresDatos/AdaptadorCamNotificaciones.java
package com.valleapp.valletpv.adaptadoresDatos;

import android.annotation.SuppressLint;
import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.valleapp.valletpv.R;
import com.valleapp.valletpv.interfaces.IControladorAutorizaciones;

import org.json.JSONObject;

import java.util.List;

public class AdaptadorCamNotificaciones extends ArrayAdapter<JSONObject> {


    private final List<JSONObject> objects;
    private IControladorAutorizaciones controlador;

    public AdaptadorCamNotificaciones(Context context, int resource,
                                      List<JSONObject> objects,
                                      IControladorAutorizaciones controlador) {
        super(context, resource, objects);
        this.objects = objects;
        this.controlador = controlador;
    }


    @Override
    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
        LayoutInflater inflater = (LayoutInflater) getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        @SuppressLint("ViewHolder") View v = inflater.inflate(R.layout.item_camarero_notificable, parent, false);
        try {
            TextView n = (TextView) v.findViewById(R.id.txt_nombre_camarero_notificaciones);
            JSONObject o = objects.get(position);
            Log.i("ADAPTADOR", o.toString());
            n.setText(o.getString("Nombre"));
            ImageView btn = (ImageView) v.findViewById(R.id.btn_send_cam_autorizado);
            btn.setTag(o.getString("ID"));
            btn.setOnClickListener(view -> {
                  controlador.pedirAutorizacion(view.getTag().toString());
            });

        }catch (Exception e){
            e.printStackTrace();
        }
        return v;
    }
}

File: android/ValleTPV/ValleTPV/src/main/java/com/valleapp/valletpv/adaptadoresDatos/AdaptadorCamNotificaciones.kt
package com.valleapp.valletpv.adaptadoresDatos

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.valleapp.valletpv.R
import com.valleapp.valletpv.interfaces.IControladorAutorizaciones
import org.json.JSONObject

class AdaptadorCamNotificaciones(
        context: Context,
        resource: Int,
        private val objects: List<JSONObject>,
        private val controlador: IControladorAutorizaciones
) : ArrayAdapter<JSONObject>(context, resource, objects) {

override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
    val viewHolder: ViewHolder
    val view: View

    if (convertView == null) {
        val inflater = LayoutInflater.from(context)
        view = inflater.inflate(R.layout.item_camarero_notificable, parent, false)

        viewHolder = ViewHolder(
                view.findViewById(R.id.txt_nombre_camarero_notificaciones),
                view.findViewById(R.id.btn_send_cam_autorizado)
        )

        view.tag = viewHolder
    } else {
        view = convertView
        viewHolder = view.tag as ViewHolder
    }

    try {
        val obj = objects[position]
        val nombre = obj.optString("nombre", "").trim()
        val apellidos = obj.optString("apellidos", "").trim()

        // Si nombre o apellidos están vacíos, se muestra "Para todos"
        viewHolder.nombre.text = if (nombre.isNotEmpty()) {
            "$nombre $apellidos"
        } else {
            "Nombre  vacio"
        }
        viewHolder.boton.tag = obj.getString("ID")
        viewHolder.boton.setOnClickListener {
            controlador.pedirAutorizacion(it.tag.toString())
        }
    } catch (e: Exception) {
        e.printStackTrace()
    }

    return view
}

private class ViewHolder(
        val nombre: TextView,
        val boton: ImageView
)
}
